src/python_testing/tc_mode_base.py

```diff
@@ -0,0 +1,139 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+from mobly import asserts
+from chip.clusters.Types import NullValue
+
+logger = logging.getLogger(__name__)
+
+# Maximum value for ModeTags according to specs is 16bits.
+MAX_MODE_TAG = 0xFFFF
+# According to specs, the specific MfgTags should be defined in the range 0x8000 - 0xBFFF
+START_MFGTAGS_RANGE = 0x8000
+END_MFGTAGS_RANGE = 0xBFFF
+
+
+class ClusterModeCheck:
```

Comment:
Base on the description:
  - This class holds common checks between different test cases
  - this is based on `ModeBase` common things

How about naming this `ModeBaseCluserChecks` (and adjust the file name as well to not contain `base`) ?

Reply:
Changed the name for the class and also the name of the file. I'm not sure if I can completely remove the word "base" since the name of the "parent" cluster is ModeBase, but hopefully it is less general now.